### ISSUE TITLE
Good url regex for custom rule? [Question]

### ISSUE THREAD

**Issue by TheoBdev** [realm/SwiftLint#1942]
Hi & thanks for this awesome tool!

I am new to regex & find it difficult to get a working regex to match URLs in strings. 
I tried with `^(http://www.|https://www.|http://|https://)?[a-z0-9]+([-.]{1}[a-z0-9]+)*.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?$` : 

```
urls_usage:
    included: ".*.swift"
    name: "Hard Coded Urls Usage"
    regex: "^(http://www.|https://www.|http://|https://)?[a-z0-9]+([-.]{1}[a-z0-9]+)*.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?$"
    match_kinds:
      - string
    message: "Use the URL struct alternative instead."
    severity: error
```

Unfortunately, this does not match any of those test strings : 

```
let test = "http://google.com"
let test1 = "https://google.com"
let test2 = "www.google.com"
let test3 = "https://www.google.com"
let test4 = "http://www.google.com"
let test6 = "https://google.io"
let test7 = "https://google.help.io"
```

What would be a good swiftlint compatible regex for this?

**Comment by TheoBdev** [comment]
We ended up settling for `regex: '\S*http(s)?\:\S+'` even if it does not catch every possible form
